### PR TITLE
MM-36881 Fix scroll perf degradation with pull to refresh enabled

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
this PR aims to fix the performance degradation on Android after bringing back the option to pull to refresh.

on iOS behavior and performance should remain the same and for Android the performance issues should not be present and the user should have the option to pull to refresh when at the bottom of the post list.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by josephbaylon — APPROVED** [review]
LGTM. User is able to pull up to refresh on Android and also no perf degradation on post list scrolling. The same goes with iOS.

**Comment by amyblais** [comment]
@anurag6713 Please help prioritize review. @svelle Added you as a reviewer to if you're able to help check whether there any Android performance issues with this PR.

File: app/components/post_list/post_list_refresh_control.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            colors={[theme.onlineIndicator, theme.awayIndicator, theme.dndIndicator]}
+            tintColor={theme.centerChannelColor}
+            onRefresh={onRefresh}
+        />
+    );
+
+    return React.cloneElement(
+        children,
+        {refreshControl, inverted: true},
+    );
```
Comment (anurag6713) [inline app/components/post_list/post_list_refresh_control.tsx:52]: Non blocking suggestion for the common props in android and iOS. 
```suggestion
    const props = {
        colors: [theme.onlineIndicator, theme.awayIndicator, theme.dndIndicator],
        onRefresh,
        refreshing,
        tintColor: theme.centerChannelColor,
    };

    if (Platform.OS === 'android') {
        return (
            <RefreshControl
                {...props}
                enabled={enabled}
                style={style.container}
            >
                {children}
            </RefreshControl>
        );
    }

    const refreshControl = (
        <RefreshControl />
    );

    return React.cloneElement(
        children,
        {refreshControl, inverted: true, ...props},
    );
```

**Review by anurag6713 — APPROVED** [review]
LGTM! 🙂

File: app/components/post_list/post_list_refresh_control.tsx (hunk 1 above)
Comment (enahum) [reply app/components/post_list/post_list_refresh_control.tsx:52]: Like it

**Comment by enahum** [comment]
Going to merge.. perhaps we should ask people to test this in the beta build

**Pull request merged**